### PR TITLE
finish add and remove using react & redux

### PR DESCRIPTION
**Pull request by majdalkilany**
connect redux to react  
remove and add books using the dispatch function  
update the UI using use selector hook

### CODE REVIEW

File: src/components/CreateBook.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
+  const [category, setCategory] = useState('');
   const handleSubmit = (event) => {
     event.preventDefault();
     const newBook = {
-      title,
-      selectedCategory,
+      title: event.target.bookTitle.value,
+      author: event.target.bookAuthor.value,
+      category,
     };
```
Comment (abdulhamiid) [inline src/components/CreateBook.js]: To add a new book, please kindly ensure that the component `CreateBook` has a local state for `title` and `author`, which will be updated through the `onChange`. 👍 

```JavaScript
  const [title, setTitle] = useState('');
  const [author, setAuthor] = useState('');
  const [category, setCategory] = useState('');
```

As your application grows, it helps to be more intentional about how your state is organized and how the data flows between your components. A redundant or duplicate state is a common source of bugs  🤗

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @majdalkilany 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Use store, actions, and reducers in React.
- ✔️ Connect React and Redux.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by majdalkilany** [comment]
add local state to create book component

**Review by Strangeal — APPROVED** [review]
Hi @majdalkilany and @abdulhamiid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Alt Text

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**